src/components/general/Blog.vue

```diff
@@ -9,12 +25,122 @@ export default {
 	props: {
 		limit: Number
 	},
+	data() {
+		return {
+			blogData: []
+		}
+	},
+	methods: {
+		toText(htmlText) {
+			let plainText = document.createElement('div')
+			plainText.innerHTML = htmlText
+			htmlText = plainText.innerText
+			return htmlText
+		}
+	},
+	beforeMount() {
+		let referenceThis = this
+		let httpRequest = new XMLHttpRequest();
+		httpRequest.onreadystatechange = function() {
+			if (this.readyState == 4 && this.status == 200) {
+				referenceThis.blogData = JSON.parse(this.responseText);
+			}
```

Comment:
To avoid `this` issues, use a lambda (arrow) function instead of the `function` key word. That should make it so you can use `this` inside the function.
`httpRequest.onreadystatechange = () => {...`